paimon-web-ui/src/views/metadata/components/table-column-content/index.tsx

```diff
@@ -60,11 +60,11 @@ export default defineComponent({
       props.modelValue.splice(i, 1)
     }
 
-    watch(props.modelValue, (newData) => {
-      newData.forEach((item, index) => {
+    watchEffect(() => {
+      props.modelValue.forEach((item, index) => {
         item.sort = index
       })
-    }, { deep: true })
+    })
```

Comment:
Why not reset the sort before submit, but need to reset it when it changes?

This will result in unnecessary rendering.

Reply:
done.